### PR TITLE
Add support for client.reconnect method in Stratum V1 protocol

### PR DESCRIPTION
**Pull request by bears-fan**
#41 

This adds support for the client.reconnect method. This allows the miner to handle client.reconnect messages sent by pools, setting the miner to reconnect to a different host and port, optionally after a specified wait time (defaults to 1 second). 

Changes:
- added fields to stratum_api.h
- added parsing the json msgs from the CLIENT_RECONNECT method
- on reconnect, close the current connection and connect to the new one
- improved socket handling by marking the socket as invalid (-1) whenever it's closed

Test instructions:

1. Setup the [mock server](https://github.com/bears-fan/mock-stratum-v1-server) and start the server with default settings `python3 server.py`
2. Point the bitaxe at the mock server
3. While the server is running, use the command `reconnect`, enter host `public-pool.io` and port `21496`
4. Follow the bitaxe logs to verify expected usage. Should see a log entry similar to `Received client.reconnect: host=public-pool.io, port=21496, wait=10
`
5. The miner should gracefully close the connection and connect after the wait time specified

### CODE REVIEW

File: main/tasks/stratum_task.c [hunk 1]
```diff
@@ -210,10 +210,33 @@ void stratum_task(void * pvParameters)
                     GLOBAL_STATE->extranonce_str = stratum_api_v1_message.extranonce_str;
                     GLOBAL_STATE->extranonce_2_len = stratum_api_v1_message.extranonce_2_len;
                 } else if (stratum_api_v1_message.method == CLIENT_RECONNECT) {
-                    ESP_LOGE(TAG, "Pool requested client reconnect...");
+                    ESP_LOGI(TAG, "Pool requested client reconnect to %s:%d (wait: %d seconds)",
+                        stratum_api_v1_message.new_host ? stratum_api_v1_message.new_host : "same host",
+                        stratum_api_v1_message.new_port ? stratum_api_v1_message.new_port : port,
+                        stratum_api_v1_message.wait_time);
+
+                    if (stratum_api_v1_message.new_host) {
```
Comment (johnny9) [inline main/tasks/stratum_task.c:218]: Should probably check that its actually a new url. Do we know if the new stratum url should persist?.

**Comment by johnny9** [comment]
Have a better commit message. Include the context/module and issue # after the body

**Comment by johnny9** [comment]
Any instructions on how to test?

**Review by johnny9 — COMMENTED** [review]
Have a better commit message. Include the context/module and issue # after the body.

Also, please include instructions on how to verify.

**Comment by johnny9** [comment]
You might need to break one more loop so that DNS resolution can happen again.

**Comment by skot** [comment]
I don;t think we want to support client.reconnect. This seems like it just makes it way too easy to hijack a miner.

**Pull request closed without merging**